### DISCUSSION TITLE
Some questions

### DISCUSSION THREAD

**Discussion by michaelspence** [NOAA-EDAB/ms-keyrun#83] [General]
Hi
Caroline, Gustav and I met this week to discuss the skill assessment study. We have the following queries and requests:

1)	In fishing mortality, are there “idealised fleets”, as in each species has its own fleet that only catches that species, or is there some shared effort between species? i.e. is the fishing effort varying for different species.
2)	Is it possible to get the maturity at length?
3)	For the age/length data for the survey, what is the effort of the survey? In the initial state the numbers at length/age are huge, whereas in the surveys they are much smaller. How is the age at length data generated? Are they conditionally independent given the Atlantis output or is there a more complicated model? Do the fish in the survey die after they are caught? How were the ages of fish determined? Was there any length stratification?
4)	Can we get the diet composition by predator at length or weight as well of at age?
5)	Is it possible to get the age/length composition for the Fishery? Similarly, how are the individual numbers at age (and length) data for the fishery generated? Are they conditionally independent given the Atlantis output or is there a more complicated model? Was there any length stratification in the ages?
6)	Is Redfish’s (Seb. mentella) lengths in inches? They appear much smaller in Atlantis than they do on Wikipedia and FishBase?
7)	What other components in the Atlantis model?
8)	Can we have by haul data for the survey please in order to get spatial information on the species? 
9)	When is the deadline for this study?

Have fun

Mike

**Comment by vbartolino** [comment]
I've no answer to most of these questions but I find them all relevant as well.
Some thoughts:
2. maturity at length would be useful. If not available, can we calculate growth from age-length data and then convert mat@age into mat@len?
3. personally, I'm not worried about the absolute numbers of age at length in the survey compared to what's in the initial state as long as effort of the surveys are consistent through time. I fully agree that the survey data (lenDist, ageDist, ageLenDist) should account for the sampling design to avoid bias in the input data 
4. diet composition by predator at length would be highly desirable for gadget as well
7. do you mean info about what makes otherfood? In that case it would be useful to help the EM and get at least an order of magnitude from the OM
9. up to the group but personally I tend to see 3-year cycle of our ToRs (2022-2024) the natural time frame

**Reply by michaelspence** [reply]
A few comments on this:

2. This could be done but the data in the real world work the other way. Maturity at length is observed (e.g. DATRAS) and then calculated to maturity at age using ALKs.

3. Agree, I'd like to know more though, particularly to do with the error structure. I believe that in the real world, the residuals in the length in the surveys are correlated. e.g. if size 40cm fish are over represented in the survey, then size 41cm will also. It is a bit of work to include this in the model, as this likelihood needs to be define and developed. In Atlantis, this likelihood has been explicitly defined and I am querying what is it? Is it realistic (at least according to my beliefs) or is it simpler, e.g., the residuals through different length classes could be independent.

7. I believe so. @gustavdelius can you elaborate?

**Reply by sgaichas** [reply]
Thanks for this excellent list of questions! With your permission, I'm going to try to address each in a separate thread below so people can find them more easily. Some of these can be done with Atlantis as an operating model, and some of them cannot and will require creative solutions that we can work out to hopefully accommodate all models.

**Reply by michaelspence** [reply]
Sure go ahead. I was going to make them all one issue but was too lazy.

**Comment by sgaichas** [comment]
1. In fishing mortality, are there “idealised fleets”, as in each species has its own fleet that only catches that species, or is there some shared effort between species? i.e. is the fishing effort varying for different species.

**Reply by sgaichas** [reply]
This depends on the Atlantis setup so there isn't an answer that is true for all Atlantis models. However, for the one we are working with, NOBA, there are currently both fleets that catch only a single species, fleets that catch multiple species, and multiple fleets that catch a species. 

As a group, we should decide whether we are modeling fleet specific fisheries. Some estimation models can do this but not others. Fortunately in the Atlantis outputs I have just made a new function that can get catch by fleet for a species. The mskeyrun dataset currently aggregates them, but this can be changed.

**Reply by michaelspence** [reply]
Excellent. Is it possible to get explicit fleet catch data?

**Reply by sgaichas** [reply]
Yes, but maybe we should discuss how much resolution you want at Friday's meeting. There are 27 fleets in the model and I'm not sure it will be useful to separate them all for our purposes.

**Comment by sgaichas** [comment]
2. Is it possible to get the maturity at length?

**Reply by sgaichas** [reply]
Unfortunately, Atlantis does not output length information directly, the units of the model output for age structured groups are numbers and weight (actually weight is in units of Nitrogen but we convert that).

Our length "data" are coming from a function in the R package `atlantisom` that estimates a length composition for each age class using the distribution of weights in that age class in the model output. 

Valerio's comment is the best solution to this I think: "maturity at length would be useful. If not available, can we calculate growth from age-length data and then convert mat@age into mat@len?"

I have already fit some von Bertalanffy growth models to the age class at length data for the Hydra model. (not all of them converged! Atlantis is a bit too real sometimes...) so this is already partway there.

**Reply by vbartolino** [reply]
>I have already fit some von Bertalanffy growth models to the age class at length data for the Hydra model. (not all of them converged! Atlantis is a bit too real sometimes...) so this is already partway there.

I've done something similar i.e. to get initial values of growth params but I had some doubts on how to deal with age-length data which are expressed with agecls rather than true ages. @sgaichas, I would be interested to know how you approached that. Thanks

**Comment by sgaichas** [comment]
3. For the age/length data for the survey, what is the effort of the survey? In the initial state the numbers at length/age are huge, whereas in the surveys they are much smaller. How is the age at length data generated? Are they conditionally independent given the Atlantis output or is there a more complicated model? Do the fish in the survey die after they are caught? How were the ages of fish determined? Was there any length stratification?

**Reply by sgaichas** [reply]
These are great questions. As general background, what we are doing is using Atlantis model *output* to generate the simulated data here. Our simulated data do not come from a real-time Atlantis run; the `atlantisom` R package was designed to make use of existing outputs offline. 

The "survey" from Atlantis is not actually implemented in the Atlantis model, it is an approximation of the true state of the Atlantis model taken from model output. Therefore, nothing the "survey" does impacts the Atlantis model: fish in the "survey" do not die in the Atlantis model (not from the survey anyway). Because most of our fishery survey catches are generally assumed not to impact the populations because they are small relative to the fisheries (they are not accounted for in most stock assessments that I am aware of), I think this assumption is ok for our purposes. 

The `atlantisom` user (me right now, but I welcome input!) designs the "survey" by selecting which Atlantis model polygons are included, which output timesteps are sampled, how much bias is in the observations (survey efficiency by species and survey selectivity by age class/true age and species),  and how much observation error is in the resulting index and age compositions. The resulting survey is therefore a partial representation of the true state of each species in the output. 

I will note that modelers in real life might have some idea of the above parameters (where and when the survey operated) but not all of them (actual survey efficiency and selectivity) so we should discuss as a group how much modeling teams should know about the survey characteristics. 

The length compositions are generated as described above from the survey sampled age compositions and weights. So I know that this is the reverse of how things are in real life: it is the numbers at age that are sampled directly from Atlantis output (we can apply age error if we want, but I have not) and the length comp is generated from them. 

The initial state currently included in `mskeyrun` was designed for the hydra model, and are not generated from the survey but from the actual Atlantis state at the start of the model run. They are independent of the survey samples. We could instead generate them from surveys, but having a nearly true initial state was our initial choice to get the model up and running. Hydra simply uses this estimate as a starting point.

**Reply by vbartolino** [reply]
thanks @sgaichas for the explanation. Can we assume that effort, selectivity and coverage of the surveys are consistent through time?

**Reply by sgaichas** [reply]
Yes, that is how we are currently implementing them.

**Reply by michaelspence** [reply]
Thanks Sarah. This is a bit clearer, however I am interested in how the observation error is created. One possible model I can see is that each species (i) and (j) combination is independently sampled, e.g.:

ln (S_{i,j}) ~ N(ln (q_{i,j}N_{i,j}),sigma^2_{i,j}),

Or is there some correlation among ages sampled, e.g.

ln (S_{i}) = MNV(ln(q_{i}N_{i}),Sigma_{i}),

where S_i=(S_i1,S_i2,....,), .. survey observations of age.

I do not mean anything to do with uncertainty in ageing fish. I mean it to do with how the true age of the fish is sampled in the survey.

I understand this is not known exactly in the real world but the models make assumptions about it and modellers beliefs about it, based on understanding of the real world and survey (Vanessa's Dirichlet paper is an example of this). If Atlantis doesn't follow the real world then these assumptions and beliefs will not be met.

**Reply by sgaichas** [reply]
Thanks Mike, got it. The observation error in ages is generated by first getting the true age comp for a species for that timestep across the specified model polygons, then a multinomial sample is taken with a user-specified effective sample size to give the observed age composition.

**Reply by michaelspence** [reply]
Thanks Sarah. It seems clearer. 

So the error in the survey biomass is independent to the age error in the age/length composition?

One example could be that the effective sample size being the same every year regardless of the amount surveyed. 

Are the are the lengths are then sampled from the length at age?

**Comment by sgaichas** [comment]
4. Can we get the diet composition by predator at length or weight as well of at age?

**Reply by sgaichas** [reply]
Yes, I have converted predator age class to lengths in the diet inputs for the Hydra model, and I have it on my list to add that dataset to `mskeyrun`. I will see if I can do that today.

**Reply by sgaichas** [reply]
New dataset `simSurveyLenDietcomp` is on the `mskeyrun sarah_wgsamsim` branch now. It gives the proportion of each prey item for each 1 cm length bin of each predator. 
This could also be expressed as 10 cm length bins or some other bin that you choose.

**Comment by sgaichas** [comment]
5. Is it possible to get the age/length composition for the Fishery? Similarly, how are the individual numbers at age (and length) data for the fishery generated? Are they conditionally independent given the Atlantis output or is there a more complicated model? Was there any length stratification in the ages?

**Reply by sgaichas** [reply]
Again excellent questions, and worth distinguishing how this works from the "survey" described above.

Unlike the "survey", the fishery is implemented within the Atlantis model. So the Atlantis modeler has control over fishery (age class) selectivity and we simply take the outputs, possibly subsampling in space or time (though I haven't here) and adding observation error (currently extremely small).

Numbers at true age are taken from the Atlantis fishery output, and lengths are generated from numbers at age class and weight at age class from Atlantis output using the same function as for surveys. So unlike in the real world, our sample is of actual ages and lengths are generated from there. Overall you can assume there is no length stratification in the ages.

Yes, the ageclass/length composition for the fishery has been added to the `mskeyrun sarah_wgsamsim` branch this week. The name of the dataset is `simFisheryAgeLencomp`. Sorry for the delay on that.

Please note that lengths are generated from age classes not the true annual Atlantis ages, so what you have in this and the `simSurveyAgeLencomp` dataset are age class and length. Atlantis age classes are mapped to true ages using information in `simBiolPar` also on the `sarah_wgsamsim` branch at present using the column `AgeperAgecl`.

**Reply by michaelspence** [reply]
> Numbers at true age are taken from the Atlantis fishery output, and lengths are generated from numbers at age class and weight at age class from Atlantis output using the same function as for surveys. So unlike in the real world, our sample is of actual ages and lengths are generated from there. Overall you can assume there is no length stratification in the ages.

Does this mean that the selectivity in Atlantis is age based and not length based?

For example is this correct:

1) total catches at age are generated by F at age
2) the lengths caught at age are sampled from length at age distributions of that age class in the ecosystem conditional on 1

**Reply by sgaichas** [reply]
Yes, to my understanding this is correct. We can double check with the Atlantis modelers but I believe NOBA fishing is implemented with age class selectivity.

**Comment by sgaichas** [comment]
6. Is Redfish’s (Seb. mentella) lengths in inches? They appear much smaller in Atlantis than they do on Wikipedia and FishBase?

**Reply by sgaichas** [reply]
For this I refer to the NOBA Atlantis model documentation [here](https://imr.brage.unit.no/imr-xmlui/handle/11250/2408609). 

Redfish were parameterized baed on literature data for the Barents Sea, so their growth may differ from in other regions or overall. Here is the growth portion of the documentation (p.32):

>In NoBa, we have chosen a longevity of 40 years. Because it is a long-lived species, the growth rates (Table 3) are small and estimated based on the weight needed to be gained each day (mg N d-1) to gain the appropriate amount of weight before moving into the next age class. However, these did not fit all age classes, and were increased by a factor of 6 for age classes 1 and 2 and decreased by a factor of 4 for the older age classes. A length-weight relationship (Eq. 1) of a=0.061 and b=2.57 is used (Benjamin Planque, pers. comm.). Weight-at-age is based on Planque and Nedreaas (2011) and ICES AFWG (2010).  

The parameterization of Redfish in the Atlantis model is balancing a lot of other factors, so it may result in outputs that look a bit different from real life. 

For me, this is ok because I think that for model performance testing purposes, Atlantis does not need to fully reflect all aspects of reality of the system, but rather produce an internally consistent dataset with the characteristics needed to challenge the estimation models. We could be calling these speciesA, speciesB, etc. But if the group decides that the Redfish are not optimal, we can always leave them out.

**Reply by gustavdelius** [reply]
I think the problem might be that the a and b parameters given in `simBiolPar` are not those used in the paragraph above. 
```
simBiolPar |> filter(Name == "Redfish") |> select(WLa, WLb)
```
gives  WLa  = 0.0085 , WLb = 3.21, not a=0.061 and b=2.57.

**Comment by sgaichas** [comment]
7. What other components in the Atlantis model?

**Reply by sgaichas** [reply]
The full NOBA model set up documentation is [here](https://imr.brage.unit.no/imr-xmlui/handle/11250/2408609) and a sensitivity analysis is [here](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0210419). A more general overview of the Atlantis model framework is [here](https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.13272).

We can definitely do what is suggested in Valerio's comment: 
>do you mean info about what makes otherfood? In that case it would be useful to help the EM and get at least an order of magnitude from the OM

But please elaborate if this was a different question.

**Comment by sgaichas** [comment]
8. Can we have by haul data for the survey please in order to get spatial information on the species?

**Reply by sgaichas** [reply]
Unfortunately, simulated data isn't available at that resolution. As noted under 3. above, the simulated survey is generated from Atlantis model output, and the most refined spatial resolution of this output is the Atlantis model polygon, most comparable to the scale of a survey stratum, but sometimes much larger. We can't approximate "hauls" in the sense of a real survey. These are the polygons in NOBA:

![image](https://github.com/NOAA-EDAB/ms-keyrun/assets/8172302/5c9f776c-3ca7-4f45-b23a-ebda3526a35b)


We should definitely discuss as a group what spatial information models need because polygon level output is possible.

**Reply by michaelspence** [reply]
Is the spatial distribution homogeneous over a whole polygon? If so, polygon level output would be helpful for mizer. We require a the spatial distributions for each species, this is often found looking at hauls from survey but polygon information would be sufficient.

**Reply by sgaichas** [reply]
Yes, distribution is homogeneous over each polygon. We can output polygon information, let's discuss what spatial/temporal resolution works best for you and others.

**Reply by gustavdelius** [reply]
Hi @sgaichas . If I understood correctly at today's meeting, for Hydra you also use spatial overlap of species to influence the predation strengths between species. Where did you get this spatial information from?

**Reply by michaelspence** [reply]
> Yes, distribution is homogeneous over each polygon. We can output polygon information, let's discuss what spatial/temporal resolution works best for you and others.

Polygon level haul information would be good. Is that available?

**Comment by sgaichas** [comment]
9. When is the deadline for this study?

**Reply by sgaichas** [reply]
Valerio is in charge of this 😀

>9. up to the group but personally I tend to see 3-year cycle of our ToRs (2022-2024) the natural time frame
